src/account/UpgradeableModularAccount.sol

```diff
@@ -564,10 +562,58 @@ contract UpgradeableModularAccount is
     // solhint-disable-next-line no-empty-blocks
     function _authorizeUpgrade(address newImplementation) internal override {}
 
-    function _checkIfValidationApplies(bytes4 selector, FunctionReference validationFunction, bool isDefault)
-        internal
-        view
-    {
+    function _checkIfValidationAppliesCallData(
+        bytes calldata callData,
+        FunctionReference validationFunction,
+        bool isDefault
+    ) internal view {
+        bytes4 outerSelector = bytes4(callData[:4]);
+
+        _checkIfValidationAppliesSelector(outerSelector, validationFunction, isDefault);
+
+        if (outerSelector == IStandardExecutor.execute.selector) {
+            (address target,,) = abi.decode(callData[4:], (address, uint256, bytes));
+
+            if (target == address(this)) {
+                // There is no point to call `execute` to recurse exactly once - this is equivalent to just having
```

Comment:
This effectively prevents `execute` make self-call, right?

Reply:
I think so, because there isn't really a scenario where execute() would do a single self call except for attempted shenanigans, because the call could just be executed regularly too